opensearch-observability/src/main/kotlin/org/opensearch/observability/model/Application.kt

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.observability.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentFactory
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.utils.stringList
+import org.opensearch.observability.ObservabilityPlugin.Companion.LOG_PREFIX
+import org.opensearch.observability.util.fieldIfNotNull
+import org.opensearch.observability.util.logger
+
+/**
+ * Application main data class.
+ *  * <pre> JSON format
+ * {@code
+ * {
+ *   "name": "Cool Application",
+ *   "description": "Application that includes multiple cool services",
+ *   "baseQuery": "source = opensearch_sample_database_flights",
+ *   "servicesEntities": [
+ *       "Payment",
+ *       "Users",
+ *       "Purchase"
+ *   ],
+ *   "traceGroups": [
+ *       "Payment.auto",
+ *       "Users.admin",
+ *       "Purchase.source"
+ *   ],
+ *   "availabilityLevels": [
+ *       {
+ *           "label": "Unavailable",
+ *           "color": "#D36086",
+ *           "condition": "when errorRate() is above or equal to 2%",
+ *           "order": "0",
+ *       }
+ *   ],
+ * }
+ * }</pre>
+ */
+
+internal data class Application(
```

Comment:
could you also add tests for this (example https://github.com/opensearch-project/trace-analytics/blob/main/opensearch-observability/src/test/kotlin/org/opensearch/observability/model/TimestampTests.kt)? maybe in another PR